### ISSUE TITLE
TypeScript typing for SeriesLegendItemClickCallbackFunction is missing some properties

### ISSUE THREAD

**Issue by ngehlert** [highcharts/highcharts#10149]
according to the definition 
```
export type SeriesLegendItemClickCallbackFunction = (this: Series, event: SeriesLegendItemClickEventObject) => void;
```

the `this` value for `plotOptions.series.events.legendItemClick` is of Chart instance, however it seems like it is actually some sort of "enriched" instance. For example it has properties like `index`, `color`, `symbol` and a lot of other calculated values like `closestPointRange`.

If needed I can provide a complete list with values that are missing.

**Comment by bre1470** [comment]
@ngehlert
Thank you for your feedback! If the properties are not part of our official documentation here: https://api.highcharts.com/highcharts/ , they are most of the time intended to be private, even though they are available on runtime. Can you specify the undocumented properties which you need?

**Comment by ngehlert** [comment]
in my case I only need index. 

Yeah I noticed that all the others look like internal ones

I'd guess index makes sense to differentiate on which series was clicked (I'm just checking if there is a different way to get this information). Color and symbol and the other values should then be fetchable from the original series definition (unless not explicitly specified and internal defaults are used)

**Comment by ngehlert** [comment]
ok quick little update, my use case currently is solveable with `this.name`. But this currently only works because the name of a series is unique. Or do you have another idea based on the current `Series` class how to properly identify the clicked series from `this.chart.series`

also on a side note the `index` property is documented for some chart types, e.g. https://api.highcharts.com/highstock/series.spline.index

**Comment by bre1470** [comment]
The `index` option is not the real index of the internal series array. This is just about the order. You can simply use `this.target` - or you should at least. I think, that this is the bug here.

**Comment by ngehlert** [comment]
hm well allright. but the `SeriesLegendItemClickEventObject` interface does not have a target, only `browserEvent: PointerEvent;` and `preventDefault: () => void;`

**Comment by bre1470** [comment]
We will fix this with v7.0.4. Until then you have to cast the event object to any and the target to series yourself like this:
```
const series = (event as any).target as Highcharts.Series;
```

**Comment by ngehlert** [comment]
allright thanks
